grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -825,23 +825,38 @@ class RequisitionService {
         return new Event(eventDate: eventDate, eventType: eventType, eventLocation: eventLocation, createdBy: currentUser)
     }
 
+    Requisition transitionRequisitionStatus(Requisition requisition, RequisitionStatus requisitionStatus, EventCode eventCode, User currentUser) {
+        requisition.status = requisitionStatus
+        Event event = createEvent(eventCode, requisition.origin, new Date(), currentUser)
+        requisition.addToEvents(event)
+        requisition.save()
+    }
+
     void triggerRequisitionStatusTransition(Requisition requisitionInstance, User currentUser) {
         Requisition.withSession {
             Requisition requisition = Requisition.get(requisitionInstance.id)
-            if (requisition.status == RequisitionStatus.PENDING_APPROVAL) {
-                if (requisition.origin.approvalRequired) {
-                    Event event = createEvent(EventCode.PENDING_APPROVAL, requisition.origin, new Date(), currentUser)
-                    requisition.addToEvents(event)
-                    requisition.status = RequisitionStatus.PENDING_APPROVAL
-                    requisition.approvalRequired = true
-                    requisition.save()
-                    return
-                }
-                Event event = createEvent(EventCode.SUBMITTED, requisition.origin, new Date(), currentUser)
-                requisition.addToEvents(event)
-                requisition.status = RequisitionStatus.VERIFYING
-                requisition.approvalRequired = false
-                requisition.save()
+            switch(requisition.status) {
```

Comment:
@drodzewicz Perhaps a comment here would be handy? Please add a note here about why we have various options for status transition here, with a link to request approvals epic ticket.